Modules/Scripted/SampleData/SampleData.py

```diff
@@ -315,14 +315,21 @@ def logMessage(self, message, logLevel=logging.INFO):
     logging.log(logLevel, message)
     slicer.app.processEvents(qt.QEventLoop.ExcludeUserInputEvents)
 
-  def setCategoryVisible(self,categoryName, visible):
-    for i in range(self.layout.count()):
-      widget = self.layout.itemAt(i).widget()
-      if hasattr(widget, "name") and widget.name == '%sCollapsibleGroupBox' % categoryName:
-        if visible:
-          widget.show()
-        else:
-          widget.hide()
+  def setCategoryVisible(self, categoryName, visible):
+    """Update visibility of a SampleData category given its name.
+    """
+    if self.isRegisteredDataSource(categoryName):
+      for i in range(self.layout.count()):
+        widget = self.layout.itemAt(i).widget()
+        if hasattr(widget, "name") and widget.name == '%sCollapsibleGroupBox' % categoryName:
+          widget.setVisible(visible)
+          return
```

Comment:
I suggest to simplify even further  to something like this:

```python
  def setCategoryVisible(self, categoryName, visible):
    """Update visibility of a SampleData category given its name.
    """
    if not self.isRegisteredDataSource(categoryName):
      return
    slicer.util.findChild(self.parent, '%sCollapsibleGroupBox' % categoryName).setVisible(visible)
```

Reply:
This is a really useful function! is there any documentation somewhere that lists all the function in slicer.util?

Reply:
See https://slicer.readthedocs.io/en/latest/developer_guide/slicer.html

Reply:
This is even shorter, is it still better to split the test and the action or is it fine in one block like that?

```python 
def setCategoryVisible(self, categoryName, visible):
    """Update visibility of a SampleData category given its name.
    """
    if self.isRegisteredDataSource(categoryName):
      slicer.util.findChild(self.parent, '%sCollapsibleGroupBox' % categoryName).setVisible(visible)
```

Reply:
It is usually more readable to split things.
```
# check, early exit if needed
# then action
```